src/ann_solo/reader.py

```diff
@@ -415,6 +804,61 @@ def _parse_spectrum_mzxml(spectrum_dict: Dict) -> MsmsSpectrum:
 
     return spectrum
 
+
+def _leading_substitute_pattern(match: re.Match) -> str:
+    """
+    Takes a match object as its argument and returns the replacement string.
+
+    Parameters
+    ----------
+    match : Match
+        Match object in the input string according to the pattern.
+
+    Returns
+    -------
+    str
+        Modified string.
+    """
+    if match.group(1) and match.group(2):
+        return '[{}]?[{}]-{:s}'.format(match.group(1), match.group(2),
+                                       match.group(3))
+    elif match.group(1):
+        return '[{}]-{}'.format(match.group(1), match.group(3))
+    else:
+        return match.group(0)
+
+
+def _mgf_seq_to_proforma(peptide: str) -> str:
```

Comment:
**nitpick:** MGF format doesn't exist. I think this might refer to MassIVE-KB format instead (i.e. the MassIVE-KB spectral library)? In that case, I would modify it accordingly to avoid future confusion.